Plugins/Rename/Rename.cpp

```diff
@@ -95,74 +110,107 @@ CNWSPlayer *Rename::player(Types::ObjectID playerId)
-        plugin.GlobalNameChange(true);
+        plugin.GlobalNameChange(true, pPlayer);
     }
 }
 
-void Rename::HookPartyInvite(Services::Hooks::CallType cType, NWNXLib::API::CNWSMessage* message, CNWSPlayer* pPlayer, unsigned char c)
+void Rename::HookPartyInvite(Services::Hooks::CallType cType, CNWSMessage*, CNWSPlayer*, unsigned char)
 {
     Rename& plugin = *g_plugin;
     if (cType == Services::Hooks::CallType::BEFORE_ORIGINAL)
     {
         //traverse the player list and replace the names of all players on the override list
-        plugin.GlobalNameChange(false);
+        plugin.GlobalNameChange(false, nullptr);
     }
     else
     {
         //And now we do it again to restore the names after the party invite
-        plugin.GlobalNameChange(true);
+        plugin.GlobalNameChange(true, nullptr);
     }
 }
 
-void Rename::GlobalNameChange(bool bOriginal)
+void Rename::GlobalNameChange(bool bOriginal, CNWSPlayer* pPlayer)
 {
     API::CServerExoAppInternal* server = Globals::AppManager()->m_pServerExoApp->m_pcExoAppInternal;
- 
-    //go through all the players and change their names or restore them based on bOriginal.
-    
     API::CExoLinkedListInternal* playerList = server->m_pNWSPlayerList->m_pcExoLinkedListInternal;
     
+    std::string lsFirstName;
+    std::string lsLastName;
+    
+    //go through all the players and change their names or restore them based on bOriginal.
     for (API::CExoLinkedListNode* head = playerList->pHead; head; head = head->pNext) 
     {
-        CNWSClient* client = reinterpret_cast<API::CNWSClient*>(head->pObject);
+        CNWSClient* client = static_cast<API::CNWSClient*>(head->pObject);
         Types::ObjectID pcObjectID = static_cast<CNWSPlayer*>(client)->m_oidNWSObject;
-        std::string lsFirstName = (bOriginal) ?  *g_plugin->GetServices()->m_perObjectStorage->Get<std::string>(pcObjectID,firstNameKey) :  *g_plugin->GetServices()->m_perObjectStorage->Get<std::string>(pcObjectID,overrideNameKey);
-        std::string lsLastName = (bOriginal) ?  *g_plugin->GetServices()->m_perObjectStorage->Get<std::string>(pcObjectID,lastNameKey) : std::string("");
+        lsFirstName = (bOriginal) ?  *g_plugin->GetServices()->m_perObjectStorage->Get<std::string>(pcObjectID,firstNameKey) :  *g_plugin->GetServices()->m_perObjectStorage->Get<std::string>(pcObjectID,overrideNameKey);
+        lsLastName = (bOriginal) ?  *g_plugin->GetServices()->m_perObjectStorage->Get<std::string>(pcObjectID,lastNameKey) : std::string("");
         CNWSCreature* pCreature = server->GetCreatureByGameObjectID(pcObjectID);
         
         if (pCreature && !lsFirstName.empty() && ( bOriginal || !pCreature->m_sDisplayName.IsEmpty()))
         {
+            if (pPlayer != nullptr || static_cast<bool>(*g_plugin->GetServices()->m_perObjectStorage->Get<int>(pcObjectID, playerNameOverrideStateKey)) //if the pPlayer pointer is null (from a party invite) or the player name override is default...
```

Comment:
Style nit, but can you please keep line width to say <120 characters. It's hard to read in diff mode, and also pretty complex to reason about. In this case I'd do:
```C
    if (pCreature && !lsFirstName.empty() && ( bOriginal || !pCreature->m_sDisplayName.IsEmpty()))
    {
        bool isDM = server->GetCreatureByGameObjectID(pPlayer->m_oidNWSObject)->m_pStats->m_bIsDM;
        bool noOverride = *g_plugin->GetServices()->m_perObjectStorage->Get<int>(pcObjectID, playerNameOverrideStateKey) != 0;
        if (pPlayer != nullptr || noOverride || isDM || pPlayer->m_oidNWSObject != pcObjectID)
        {
            if (bOriginal)
```

And now that I've written it out like that, I see:
```C
if (pPlayer != nullptr || ... || pPlayer->m_oidNWSObject != pcObjectID)
```
That's by definition invalid. The only time the OIDs can be compared is if `pPlayer == nullptr`. Because, if pPlayer is not null, the checks will end early and you'll enter the if. But if `pPlayer == nullptr`, your check for OIDs is invalid, because `pPlayer->m_oidNWSObject` will crash due to dereferencing a null pointer.